control-plane/subcommand/common/common.go

```diff
@@ -86,71 +85,83 @@ func ValidateUnprivilegedPort(flagName, flagValue string) error {
 	return nil
 }
 
+// LoginParams are parameters used to log in to consul.
+type LoginParams struct {
+	// AuthMethod is the name of the auth method.
+	AuthMethod string
+	// Datacenter is the datacenter for the login request.
+	Datacenter string
+	// Namespace is the namespace for the login request.
+	Namespace string
+	// BearerTokenFile is the file where the bearer token is stored.
+	BearerTokenFile string
+	// TokenSinkFile is the file where to write the token received from Consul.
+	TokenSinkFile string
+	// Meta is the metadata to set on the token.
+	Meta map[string]string
+
+	// numRetries is only used in tests to make them run faster.
+	numRetries uint64
+}
+
 // ConsulLogin issues an ACL().Login to Consul and writes out the token to tokenSinkFile.
 // The logic of this is taken from the `consul login` command.
-func ConsulLogin(client *api.Client, cfg *api.Config, authMethodName, datacenter, namespace, bearerTokenFile, serviceAccountName, tokenSinkFile string, meta map[string]string, log hclog.Logger) (string, error) {
+func ConsulLogin(client *api.Client, params LoginParams, log hclog.Logger) (string, error) {
```

Comment:
Oh that's an interesting idea! My only reservation is that `ConsulLogin` doesn't make so much sense as a method on `LoginParams`. Something like `params.ConsulLogin(client, logger)` reads a bit counter-intuitive to me because I'm not seeing params as a struct that typically has a lot of methods as it's usually something you pass around. WDYT?

Reply:
hmm. we could add the api.Client to the struct, call it LoginHelper (name needs to be workshopped), and then have Login on that struct? It would feel more cohesive then IMO

Reply:
I think I'm going to keep it as is for now. I thought about adding maybe a sort of a wrapped consul client but then it wouldn't make sense to use it just for login and maybe we'd want to re-use it for something else, in which case it also doesn't make sense to add params to that struct. It might be a bigger rewrite also.